src/multiselect_contextmenu.js

```diff
@@ -702,30 +749,28 @@ const registerSelectAll = function() {
              !block.isInsertionMarker();
     },
     callback: function(scope) {
-      const blockSelection = blockSelectionWeakMap.get(scope.workspace);
-      if (Blockly.getSelected() &&
-        !blockSelection.has(Blockly.getSelected().id)) {
+      const multiDraggable = multiDraggableWeakMap.get(scope.workspace);
+
+      // Make sure that there is nothing in the multiDraggable (clearing)
+      // prior to selecting all blocks in workspace.
+      if (Blockly.getSelected()) {
         Blockly.getSelected().pathObject.updateSelected(false);
         Blockly.common.setSelected(null);
+        multiDraggable.clearAll();
       }
       const blockList = [];
-      scope.workspace.getTopBlocks().forEach(function(block) {
+      scope.workspace.getAllBlocks().forEach(function(block) {
```

Comment:
Why do you change `getTopBlocks()` into `getAllBlocks()` here? Is there a specific reason for this change? If not, I would suggest we revert the change here, since getting the top blocks is already enough for Dragging/Copy/Duplicate, also this might cause the page to crash easily if there are too many blocks on one page. Also, this is for the consistency with the old version.